include/swift/Runtime/Metadata.h

```diff
@@ -2638,28 +2659,58 @@ swift_getGenericWitnessTable(GenericWitnessTable *genericTable,
 /// \brief Fetch a uniqued metadata for a function type.
 SWIFT_RUNTIME_EXPORT
 const FunctionTypeMetadata *
-swift_getFunctionTypeMetadata(const void *flagsArgsAndResult[]);
+swift_getFunctionTypeMetadata(FunctionTypeFlags flags,
+                              const Metadata **parameters,
+                              const uint32_t *parameterFlags,
+                              const Metadata *result);
 
 SWIFT_RUNTIME_EXPORT
 const FunctionTypeMetadata *
 swift_getFunctionTypeMetadata1(FunctionTypeFlags flags,
-                               const void *arg0,
-                               const Metadata *resultMetadata);
+                               const Metadata *arg0,
+                               const Metadata *result);
+
+SWIFT_RUNTIME_EXPORT
+const FunctionTypeMetadata *
+swift_getFunctionTypeMetadata1WithFlags(FunctionTypeFlags flags,
+                                        const Metadata *arg0,
+                                        const uint32_t flags0,
+                                        const Metadata *result);
 
 SWIFT_RUNTIME_EXPORT
 const FunctionTypeMetadata *
 swift_getFunctionTypeMetadata2(FunctionTypeFlags flags,
-                               const void *arg0,
-                               const void *arg1,
-                               const Metadata *resultMetadata);
+                               const Metadata *arg0,
+                               const Metadata *arg1,
+                               const Metadata *result);
 
 SWIFT_RUNTIME_EXPORT
 const FunctionTypeMetadata *
-swift_getFunctionTypeMetadata3(FunctionTypeFlags flags,
-                               const void *arg0,
-                               const void *arg1,
-                               const void *arg2,
-                               const Metadata *resultMetadata);
+swift_getFunctionTypeMetadata2WithFlags(FunctionTypeFlags flags,
+                                        const Metadata *arg0,
+                                        const uint32_t flags0,
```

Comment:
Don't const-qualify a parameter like this.  Also, this should be ParameterTypeFlags, right?

Reply:
Got it!

Reply:
I think it's fine to make it `ParameterFlags` for 1/2/3 functions but should it be `const ParameterFlags *` for the general case too?

Reply:
The way I see is it that the use of a target-specific type is specifically just for passing these values as direct arguments to the runtime functions.  Arguably that implies that we should make the enums (``{Function,Parameter}TypeFlags``) non-templated and just make the runtime functions take a size_t with a comment about what enum they really are.  The place where we take an array of them by reference should definitely use a pointer to a 32-bit type.

Reply:
Alright, this means that my changes are correct :)